app/models/game.rb

```diff
@@ -24,6 +24,18 @@ def populate_board!
     end
   end
 
+  def stalemate?(color)
+    Piece.where(game: self).where.not(color: color).each do |piece|
```

Comment:
Huzzah! Removing the .not got the stalemate notices to flash for me. Will work on incorporating the check for a piece in the spot that it's moving to and the other bits you mentioned.